### ISSUE TITLE
Get xarray aggregation tests working

### ISSUE THREAD

**Issue by agstephens** [roocs/daops#8]
* daops version: *
* Python version: 3.7+
* Operating System: linux

### Description

@ellesmith88: I have created the following unit test module:

https://github.com/roocs/daops/blob/master/tests/test_xarray/test_xarray_aggregation.py

Most of it is in the form of skeleton code/stubs. Please can you get it working as a valid unit test. 

The purpose of it is to ensure that we have tested the _normal_ behaviour of `xarray.open_mfdataset()` - just to make sure that our assumptions throughout `roocs` are appropriate.

**Comment by ellesmith88** [comment]
@agstephens I've looked at these tests and found that introducing these changes doesn't cause the aggregation to break. The tests show that if the change is made on the first file in the time series then the new netcdf file includes the change otherwise the change is ignored.

https://github.com/roocs/daops/blob/issue/8/xarray_agg_tests/tests/test_xarray/test_xarray_aggregation.py

If the change is made to all 3 there is an error - I'm still looking at this and making the files temporary.

**Comment by agstephens** [comment]
@ellesmith88: Do the statements above relate to all the different types of changes that are tested in `test_xarray_aggregation.py`?

**Comment by ellesmith88** [comment]
@agstephens Yes it does, just added a second test for changing `var_id` to show this.

**Comment by agstephens** [comment]
Thanks @ellesmith88, are there any arguments that we can set to force the behaviour to be stricter?

**Comment by ellesmith88** [comment]
@agstephens Sorry I just realised this isn't the case for `var_id` which actually creates a new variable so that `new_var_id` and `old_var_id` are both in `ds.variables` no matter which file is modified.

Looking at the arguments - haven't found anything that works yet.

**Comment by agstephens** [comment]
@ellesmith88: that seems like a logical response, that `xarray` would just load two separate variables.
